### PR TITLE
Adddlae2

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

File: native/dlae2.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	if adf > ab {
+		rt = adf * math.Sqrt(1+(ab/adf)*(ab/adf))
+	} else if adf < ab {
+		rt = ab * math.Sqrt(1+(adf/ab)*(adf/ab))
+	} else {
+		rt = ab * math.Sqrt2
+	}
+	if sm < 0 {
+		rt1 = 0.5 * (sm - rt)
+		rt2 = (acmx/rt1)*acmn - (b/rt1)*b
```
Comment (kortschak) [inline native/dlae2.go]: Can you explain the fortran comment: "Order of execution important. To get fully accurate smaller eigenvalue, next line needs to be executed in higher precision."

File: native/dlae2.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	}
+	var rt float64
+	if adf > ab {
+		rt = adf * math.Sqrt(1+(ab/adf)*(ab/adf))
+	} else if adf < ab {
+		rt = ab * math.Sqrt(1+(adf/ab)*(adf/ab))
+	} else {
+		rt = ab * math.Sqrt2
+	}
+	if sm < 0 {
```
Comment (kortschak) [inline native/dlae2.go]: I think this set of conditions can be more nicely expressed in Go, but we'll leave it for the progressive clean-up later.

**Comment by kortschak** [comment]
LGTM

File: native/dlae2.go (hunk 1 above)
Comment (btracey) [inline native/dlae2.go]: No, but I believe is is that you have to do the divisions before the multiplications. This happens automatically with Go parenthesis rules. I would imagine the next means to do it in 80-bit precision if you can. This happens automatically with the Go spec rules (as we aren't explicitly rounding with a float64())

**Pull request merged**